### DISCUSSION TITLE
ToolbarAction and options

### DISCUSSION THREAD

**Discussion by GandonNicolas** [sulu/sulu#6794] [Q&A] [answered]
Hello, 

Is there a complete list of `ToolbarAction` and `options` available options somewhere?

**Comment by niklasnatter** [comment, marked as answer]
Hey, unfortunately this is not documented in the official documentation yet 😕 
But if you add a toolbar action with a unknown type, Sulu will log an error that lists all available types to the browser console. On 2.5.1, the error reads like this:

```
Error: There is no toolbar item with key "invalid-type" registered!

Registered keys: sulu_admin.copy, sulu_admin.copy_locale, sulu_admin.delete, sulu_admin.delete_draft, sulu_admin.dropdown, sulu_admin.publish, sulu_admin.save, sulu_admin.save_with_form_dialog, sulu_admin.save_with_publishing, sulu_admin.set_unpublished, sulu_admin.toggler, sulu_admin.type, sulu_security.enable_user
```

To learn more about a specific toolbar action, you can use the GitHub search to find the implementation. For example, search for `sulu_admin copy_locale`: https://github.com/sulu/sulu/search?q=sulu_admin+copy_locale
I know this is not great, but it should hopefully help you out until this is included in the documentation.
